src/shogun/distance/EuclideanDistance.h

```diff
@@ -112,6 +117,21 @@ class CEuclideanDistance: public CRealDistance
 		 *  @return distance value or upper_bound
 		 */
 		virtual float64_t distance_upper_bounded(int32_t idx_a, int32_t idx_b, float64_t upper_bound);
+		
+		/**
+		 * Precompute squared norms from features of right hand side
+		 */
+		virtual void precompute_rhs_squared_norms();
+
+		/**
+		 * Precompute squared norms from features of left hand side
+		 */
+		virtual void precompute_lhs_squared_norms();
+	
+		/**
+		 * Reset squared norms for features of both sides
+		 */
+		virtual void reset_squared_norms();
```

Comment:
Should contain instructions that this needs to be called in case of user changes the features